api/envoy/config/route/v3/route_components.proto

```diff
@@ -378,6 +378,13 @@ message WeightedCluster {
     // :ref:`FilterConfig<envoy_v3_api_msg_config.route.v3.FilterConfig>`
     // message to specify additional options.]
     map<string, google.protobuf.Any> typed_per_filter_config = 10;
+
+    oneof host_rewrite_specifier {
+      // Indicates that during forwarding, the host header will be swapped with
+      // this value.
+      string host_rewrite_literal = 11
+          [(validate.rules).string = {well_known_regex: HTTP_HEADER_VALUE strict: false}];
+    }
```

Comment:
Seems fine, but do we anticipate much more substantial growth here? I take it this is mostly a byproduct of stricter restrictions on what can be set via `response_headers_to_add` etc. Should the rationale for using this be stated more explicitly in docs?

Reply:
your call - as with all Envoy config I think we add it as needed.  I think it's reasonable to allow host/authority/scheme rewrites anywhere we allow request/response header mutations, but no need to clutter up configs/code adding them where they're not needed?

Reply:
I was chatting with @stevenzzzz about opportunities to save memory on repeated fields in `RouteAction` today. Using hierarchy is one blunt tool for this, so we may want more fields available for setting at `VirtualHost` and then overrriding at different levels. But, we'll need to confront the existing prior art with request/response header modifications, so I think we can land this PR independent of that.